src/gabarit_store.erl

```diff
@@ -1,19 +1,477 @@
+  case get_template_version_impl(TemplateId, Version) of
+    {ok, Content} ->
+      gen_statem:reply(From, {ok, Content});
+    {error, Reason} ->
+      gen_statem:reply(From, {error, Reason})
+  end,
+  {keep_state, State};
+
+locked({call, From}, list_namespaces, State) ->
+  Namespaces = ets:tab2list(?NAMESPACE_TABLE),
+  gen_statem:reply(From, {ok, [{N, L} || {N, L, _} <- Namespaces]}),
+  {keep_state, State};
+
+locked({call, From}, {list_templates, Namespace}, State) ->
+  Templates = ets:match_object(?TEMPLATE_TABLE, {'_', Namespace, '_', '_'}),
+  gen_statem:reply(From, {ok, [{Id, Path} || {Id, _, Path, _} <- Templates]}),
+  {keep_state, State};
+
+locked({call, From}, {list_versions, TemplateId}, State) ->
+  Versions = ets:match_object(?VERSION_TABLE, {{TemplateId, '_'}, '_', '_', '_'}),
+  gen_statem:reply(From, {ok, [{V, T} || {{_, V}, _, _, T} <- Versions]}),
+  {keep_state, State};
+
+locked(_EventType, _EventContent, State) ->
+  {keep_state, State}.
+
+%%====================================================================
+%% Helper Functions
+%%====================================================================
+
+%% @doc Checks if a namespace has reached its template limit.
+%% @param Namespace The namespace to check
+%% @returns {ok, CurrentCount} | {error, Reason}
+%% @private
+check_namespace_limit(Namespace) ->
+  case ets:lookup(?NAMESPACE_TABLE, Namespace) of
+    [{Namespace, Limit, Count}] when Count < Limit ->
+      {ok, Count};
+    [{_, _, _}] ->
+      {error, namespace_limit_reached};
+    [] ->
+      {error, namespace_not_found}
+  end.
+
+%% @doc Generates a unique template identifier.
+%% @returns TemplateId :: integer()
+%% @private
+generate_template_id() ->
+  erlang:phash2({node(), self(), erlang:system_time(), make_ref()}).
+
+%% @doc Compresses content if it exceeds the specified threshold.
+%% @param Content The content to potentially compress
+%% @param Threshold Size threshold for compression in bytes
+%% @returns {CompressedContent, IsCompressed}
+%% @private
+compress_if_needed(Content, Threshold) ->
+  if byte_size(Content) > Threshold ->
+       {zlib:compress(Content), true};
+     true ->
+       {Content, false}
+  end.
+
+%% @doc Decompresses content if it was previously compressed.
+%% @param Content The potentially compressed content
+%% @param IsCompressed Boolean indicating if content is compressed
+%% @returns DecompressedContent
+%% @private
+decompress_if_needed(Content, true) ->
+  zlib:uncompress(Content);
+decompress_if_needed(Content, false) ->
+  Content.
+
+%% @doc Performs cleanup of old template versions.
+%% Keeps a configurable number of the most recent versions.
+%% @returns ok
+%% @private
+perform_cleanup() ->
+  Templates = ets:tab2list(?TEMPLATE_TABLE),
+
+  MaxVersionsToKeep = application:get_env(gabarit, max_versions_to_keep, 5),
+
+  lists:foreach(fun({TemplateId, _Namespace, _Path, CurrentVersion}) ->
```

Comment:
Instead of creating a lambda function, create a full function with a bit of documentation. It will be easier to read.

```erlang
perform_cleanup_foreach({TemplateId, Namespace, Path, CurrentVersion}) ->
  % ...
  ok.

perform_cleanup() ->
  lists:foreach(fun perform_cleanup_foreach/1,
  % ...
```